time_stamper_cv/include/convex_shape.h

```diff
@@ -0,0 +1,112 @@
+#pragma once
+#include <vector>
+#include <opencv2/core/types.hpp>
+#include "common.h"
+
+/**
+ * This class is used to detect a given convex shape. First, a convex hull is calculated out of all keypoints.
+ * Thereupon, it calculates if the convex hull match the given convex shape.
+ *
+ * At the moment only hardcoded shape (@var virtualCorners) is supported.
+ * ConvexShape should support all polygons in future.
+ */
+
+class ConvexShape {
+ public:
+  /**
+   * Creates ConvexShape object
+   * @param tolerance is used for isInRange().
+   */
+  explicit ConvexShape(int tolerance);
+
+  /**
+   * Calculates a convex shape from given points.
+   *
+   * @param raw_points
+   */
+  void process(const PointVector& raw_points);
+
+  /**
+   * Get all points which define the convex shape.
+   * @return
+   */
+  PointVector getHull() const;
+
+  /**
+   * Checks if hull is valid.
+   * @return true if hull has expected size, otherwise false.
+   */
+  bool isHullValid() const;
+
+  /**
+   * Used to get scaled virtualCorners.
+   * @param scaling
+   * @return Scaled virtualCorners
+   */
+  Point2fVector getVirtualCorners(float scaling = 1.0f);
```

Comment:
param const

Reply:
Added const in cpp file to avoid warnings